NotesApplication/Screens/Editor/NotesEditorViewController.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  NotesEditorViewController.swift
+//  NoteApp
+//
+//  Created by Ana Ioramashvili on 02.11.24.
+//
+
+import UIKit
+
+protocol NotesEditorDelegate: AnyObject {
+    func didSaveNote(note: Note)
+    func didUpdateNote(note: Note, at index: Int)
+}
+
+final class NotesEditorViewController: UIViewController {
+    // MARK: - Properties
+    weak var delegate: NotesEditorDelegate?
+    private let viewModel: NotesEditorViewModel
+    private var isEditingMode: Bool = false
+    
+    // MARK: - UI Components
+    private let titleTextView: UITextView = {
+        let textView = UITextView()
+        textView.font = .systemFont(ofSize: 28, weight: .bold)
+        textView.textColor = .white
+        textView.backgroundColor = .clear
+        textView.textContainer.maximumNumberOfLines = 0
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        textView.isScrollEnabled = false
+        return textView
+    }()
+    
+    private let contentTextView: UITextView = {
+        let textView = UITextView()
+        textView.font = .systemFont(ofSize: 16)
+        textView.backgroundColor = .clear
+        textView.textColor = .white
+        textView.textContainer.maximumNumberOfLines = 0
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        return textView
+    }()
+    
+    // MARK: - Initialization
+    init(note: Note? = nil, index: Int? = nil) {
+        self.viewModel = NotesEditorViewModel(note: note, index: index)
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+        setupNavigationBar()
+        setupDelegates()
+        
+        if viewModel.isEditingMode {
```

Comment:
Remove logic from viewDidLoad(). Make a separate function that is called in setupUI as the logic below is controlling how UI behaves.